src/common/utils/csv-download-utils.ts

```diff
@@ -0,0 +1,80 @@
+import { GetTransactionListTypeEnum } from '@stacks/blockchain-api-client';
+import {
+  AddressBalanceResponse,
+  AddressTransactionsWithTransfersListResponse,
+  AddressTransactionWithTransfers,
+} from '@stacks/stacks-blockchain-api-types';
+import dayjs from 'dayjs';
+import { InfiniteData, QueryClient } from 'react-query';
+import { microStxToStx, microToStacks, toRelativeTime } from '../utils';
+
+export type CSVDownloadObjectType = {
+  Date: string;
+  'Sent Amount': string | number;
+  'Sent Currency': string;
+  'Received Amount': string | number;
+  'Received Currency': string;
+  'Fee Amount': string | number;
+  'Fee Currency': string;
+  'Net Worth Amount': string | number;
+  'Net Worth Currency': string;
+  TxHash: string;
+};
+
+export const getTxnData = (
+  queryClient: QueryClient,
+  activeFilters: Record<GetTransactionListTypeEnum, boolean>,
+  address: string
+): CSVDownloadObjectType[] => {
+  const txsQueryData = queryClient.getQueriesData<
+    InfiniteData<AddressTransactionsWithTransfersListResponse>
+  >({
+    queryKey: ['addressConfirmedTxsWithTransfersInfinite', address],
+  });
+
+  const addressBalanceQueryData = queryClient.getQueriesData<AddressBalanceResponse>({
+    queryKey: ['accountBalance', address],
+  });
+
+  const balance = addressBalanceQueryData[0][1].stx.balance;
+
+  const txs = txsQueryData[0][1].pages.reduce<AddressTransactionWithTransfers[]>(
+    (acc, { results }) => {
+      return acc.concat(results);
+    },
+    []
+  );
+
+  const filteredTxs = txs?.filter(tx => activeFilters[tx.tx.tx_type]);
+  return formatTxns(filteredTxs, balance);
+};
+
+const formatTxns = (
+  txs: AddressTransactionWithTransfers[],
+  balance: string | number
+): CSVDownloadObjectType[] => {
+  let currentNetWorth = Number(balance);
```

Comment:
I suggest renaming this to `stxBalance`, let me know what you think!

Reply:
Sure

Reply:
In this case, we can keep the name, thank you!